pkg/autoscaler/controller/laddercontroller/ladder_controller.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package laddercontroller
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"sort"
+
+	"github.com/kubernetes-incubator/cluster-proportional-autoscaler/pkg/autoscaler/controller"
+
+	"github.com/golang/glog"
+)
+
+var _ = controller.Interface(&LadderController{})
+
+const (
+	ControllerType = "ladder"
+)
+
+type LadderController struct {
+	params  *ladderParams
+	version string
+}
+
+func NewLadderController() controller.Interface {
+	return &LadderController{}
+}
+
+type paramEntry [2]int
+
+type paramEntries []paramEntry
+
+func (entries paramEntries) Len() int {
+	return len(entries)
+}
+
+func (entries paramEntries) Less(i, j int) bool {
+	return entries[i][0] < entries[j][0]
+}
+
+func (entries paramEntries) Swap(i, j int) {
+	entries[i], entries[j] = entries[j], entries[i]
+}
+
+type ladderParams struct {
+	CoresToReplicasEntries paramEntries `json:"cores_to_replicas_entries"`
```

Comment:
json names should be styled as `coresToReplicas`

having "entries" in the name is redundant.